src/main/java/com/dev/cinema/dao/impl/MovieSessionDaoImpl.java

```diff
@@ -0,0 +1,62 @@
+package com.dev.cinema.dao.impl;
+
+import com.dev.cinema.dao.MovieSessionDao;
+import com.dev.cinema.exeptions.DataProcessingException;
+import com.dev.cinema.library.Dao;
+import com.dev.cinema.model.MovieSession;
+import com.dev.cinema.util.HibernateUtil;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.util.List;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Predicate;
+import javax.persistence.criteria.Root;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+
+@Dao
+public class MovieSessionDaoImpl implements MovieSessionDao {
+    @Override
+    public List<MovieSession> findAvailableSessions(Long movieId, LocalDate date) {
+        LocalDateTime startDay = date.atStartOfDay();
+        LocalDateTime endDay = date.atTime(LocalTime.MAX);
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            CriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
+            CriteriaQuery<MovieSession> criteriaQuery =
+                    criteriaBuilder.createQuery(MovieSession.class);
+            Root<MovieSession> movieSessionRoot = criteriaQuery.from(MovieSession.class);
+            Predicate movie = criteriaBuilder.equal(movieSessionRoot.get("movie"), movieId);
+            Predicate x =
+                    criteriaBuilder.between(movieSessionRoot.get("showTime"), startDay, endDay);
+            Predicate predicate = criteriaBuilder.and(movie, x);
+            criteriaQuery.select(movieSessionRoot).where(predicate);
+            return session.createQuery(criteriaQuery).getResultList();
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't find Available Sessions", e);
+        }
+    }
+
+    @Override
+    public MovieSession add(MovieSession movieSession) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(movieSession);
+            transaction.commit();
+            return movieSession;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't add MovieSession entity", e);
```

Comment:
let's make message more informative and add some info about entity we were trying to save same with other methods, where applicable

Reply:
done